### ISSUE TITLE
🐛 [Bug report]: Puppetter Build fails intermittently: Navigation Timeout Exceeded

### ISSUE THREAD

**Issue by atapas** [reactplay/react-play#320]
**Describe the bug**
The Puppetter Error fails with this message: error at /plays TimeoutError: Navigation Timeout Exceeded: 30000ms exceeded

![image](https://user-images.githubusercontent.com/3633137/174231117-f08f66d2-1027-4a19-9264-e9be1b85f72e.png)

We need a way to configure this time out limit

**Comment by erayalkis** [comment]
As far as I know, there's currently no way to directly configure the time out limit for puppeteer, though I'll look into a workaround for this and keep you updated!

**Comment by atapas** [comment]
> As far as I know, there's currently no way to directly configure the time out limit for puppeteer, though I'll look into a workaround for this and keep you updated!

What if we remove Puppetter altogether? @erayalkis

**Comment by erayalkis** [comment]
I think the same problem existed before we installed a downgraded version of puppeteer, so I assume the problem must be somewhere else. We could maybe try downgrading react-snap?

**Comment by atapas** [comment]
Ok @erayalkis 

So we added react snap to create server side rendering or to help us with meta info? What will we loose if we remove react snap altogether?

**Comment by erayalkis** [comment]
React-snap works similarly to SSR, but it isn't exactly the same
When we create a production build, react-snap goes through each route and creates a static HTML file for it, improving performance and enabling access to the page without JavaScript! This is crucial for our meta tags to work, as popular websites (Facebook, Reddit, etc.) don't enable JavaScript when fetching meta tags to embed data.

Basically, if we removed react-snap our dynamic meta tags would no longer work. We could delete react-snap and set up SSR instead, if that's a viable solution!

**Comment by atapas** [comment]
Thank you @erayalkis for the details.

Two things I've observed.

- It fails intermittently.
- Somehow it fails more on vercel than netlify.

We do not need the dynamic meta tag functionality for the branch build. We need it only for the main production build. Our branch builds are from vercel and production build is from netlify.

I've now bypassed the post build for branch builds so that it builds fast and does not fail frequently. The main production build will still build the dynamic meta tag.

This is a makeshift approach until we find a better solution if the issue. We should continue searching for a better solution. There might be some settings we need to research on.
